styleguide/js/zoolander.js

```diff
@@ -126,6 +128,114 @@ jQuery(document).ready(function($) {
+    var $rsWindow = $(window);
+    var isFixed = false;
+    // var inBounds = false;
+
+    // Callback to get new form width on resizing/scrolling
+    function getWidth(bar, padding) {
+      var width = parseInt(bar.outerWidth(true), 10) - padding;
+      return width;
+    }
+
+    // Callback to check if in bounds of side menu
+    function inBounds(wt) {
+      return $stickyItemTop - $mainNavHeight < wt &&
+        $filterBar.outerHeight() + $filterBar.offset().top +
+        $mainNavHeight - ($filterForm.outerHeight() * 3) >
+        (wt + $filterForm.outerHeight())
+    }
+
+    $filterHamburger.click(function() {
+      $filterForm.slideToggle(150);
+    });
+
+    $rsWindow.scroll(function() {
+      // If not in mobile
+      if ($rsWindow.outerWidth() > 767) {
+        var $rsWindowTop = $rsWindow.scrollTop();
+        // If user scrolls to filter form and is in the bounds of the bar, move form up and down
+        if (inBounds($rsWindowTop)) {
+          $filterForm.css({
+            'position' : 'fixed',
+            'top' : $mainNavHeight + 10,
+            'width' : getWidth($filterBar, $barPadding) + 'px'
+          });
+          isFixed = true;
+        } else {
+          $filterForm.css({
+            'position' : '',
+            'width' : ''
+          });
+          isFixed = false;
+        }
+      }
+    });
+
+    // Set a slight delay to the resize functionality so that it doesn't
+    // run on every pixel of a browser resize
+    var resizeTimer;
+    $rsWindow.resize(function() {
+      clearTimeout(resizeTimer);
+      resizeTimer = setTimeout(function() {
+      var $rsWindowTop = $rsWindow.scrollTop();
+        // If not mobile
+        if ( $rsWindow.outerWidth() > 768 ) {
+          if (!isFixed && inBounds($rsWindowTop)) {
+            $filterForm.css({
+              'position' : 'fixed',
+              'width' : getWidth($filterBar, $barPadding) + 'px'
+            });
+            isFixed = true;
+          } else if (isFixed && inBounds($rsWindowTop)) {
+            $filterForm.css({
+              'width' : getWidth($filterBar, $barPadding) + 'px'
+            });
+          } else if (isFixed && !inBounds($rsWindowTop)) {
+            $filterForm.css({
+              'position' : ''
+            });
+            isFixed = false
+          }
+        } else {
+          // If in mobile
+          $filterForm.css({
+            'width' : ''
+          });
+          if (isFixed) {
+            $filterForm.css({
+              'position' : '',
+              'width' : ''
+            });
+            isFixed = false;
+          }
+        }
+        // Make sure form is always shown in appropriate viewport
+        // given the users interactions with browser resize.
+        if (!$filterForm.is(':visible') && $rsWindow.outerWidth() > 768) {
+          $filterForm.show();
+        } else if ($filterForm.is(':visible') && $rsWindow.outerWidth() < 768) {
+          $filterForm.hide();
+        }
```

Comment:
Is this hide/show functionality here even necessary? Seems like this CSS controls this behavior already: 

```
.rsFilter-form {
  display: none;
}

@media (min-width: $screen-sm-min) {
  .rsFilter-form {
    display: block; 
  }
}
```

Additionally by letting css handle it, instead of inside this 250ms delayed JS, we no longer have that random glitch where the sidebar just randomly disappears.